src/boosting/goss.hpp

```diff
@@ -97,18 +70,23 @@ class GOSS: public GBDT {
       is_use_subset_ = true;
     }
     // flag to not bagging first
-    bag_data_cnt_ = num_data_;
+    bag_data_cnt_ = num_data_; 
   }
 
-  data_size_t BaggingHelper(data_size_t start, data_size_t cnt, data_size_t* buffer) override {
+  void ResetConfig(const Config* config, bool is_change_dataset, 
+          std::vector<score_t, Common::AlignmentAllocator<score_t, kAlignedSize>>& gradients, 
+          std::vector<score_t, Common::AlignmentAllocator<score_t, kAlignedSize>>& hessians) override {}
```

Comment:
>hessians) override {} [](http://example.com/codeflow?start=83&length=21)

so this function is a real empty one? Or you guys just want to implement it later? Besides, since you have parameters but not used, you can use std::ignore for them to avoid compiler warnings

Reply:
We've refactor the code and move `ResetConfig` into `ResetSampleConfig`. It is no longer empty.